python/lesson03/solution/hello.py

```diff
@@ -33,9 +33,13 @@ def http_get(port, path, param, value):
     headers = {}
     tracer.inject(span, Format.HTTP_HEADERS, headers)
 
-    r = requests.get(url, params={param: value}, headers=headers)
-    assert r.status_code == 200
-    return r.text
+    try:
```

Comment:
it might be simpler doing `with span:`, which will automatically catch the exception and log it to the span.

Reply:
Let me read more on this syntax; Will update the code accordingly or ask for help, if needed.

Reply:
@yurishkuro  I tried following the `with` syntax in python code with below code snippet:

```
with span:
        r = requests.get(url, params={param: value}, headers=headers)
        assert r.status_code == 200
        return r.text
```

It does log error messages, but it somehow reports the same span twice to Jaeger. I am a noob in python and couldn't debug more on this.

I tried removing the code and reverting back to the original state. The older code was in-fact marking errors on the span with `error = 'True'` tag. However, I couldn't see it in the UI. I suspect only capital 'T' in the True flag that could be causing this.

Let me know if I am missing something.

Reply:
`with span:` will finish the span on exit from the block. It is the recommended way of using spans, but you can only use it once for a given span. 

I see that it is already used in the callers as `with scope:`, so you should not repeat it here. It's quite possible that you don't need to add anything here - try with the code as is, I bet it will already tag the span with error as needed.

Reply:
It tags the span with `error='True'` flag, but error is not highlighted in the UI. I suspect it could be because of either my environment setup or python-jaeger-client library.

I am reverting python changes in this PR for now.

Reply:
@yurishkuro did you find the issue in python code? I would like to know if I was missing something.

Reply:
@yurishkuro Should I raise an issue for this?

Reply:
do you have a screenshot of span tags?

Reply:
<img width="1680" alt="Screenshot 2020-07-18 at 12 50 28 PM" src="https://user-images.githubusercontent.com/8849040/87847390-6e391980-c8f5-11ea-9972-63df68713892.png">

Reply:
<img width="1680" alt="Screenshot 2020-07-18 at 12 52 26 PM" src="https://user-images.githubusercontent.com/8849040/87847404-90cb3280-c8f5-11ea-882c-7e8d7c9996a4.png">

Reply:
Thanks. I think it's the uppercase in `error="True"` that's confusing the UI. There are two ways to fix that:
  1. make sure that the Python client converts Boolean tags into proper Boolean value in Thrift
  2. add a [sanitizer][] to Jaeger collector that converts error tags with string values into Boolean tags. I always thought we had one, but looks like it's only for [Zipkin spans][2]

Would you like to create a pull request?

[sanitizer]: https://github.com/jaegertracing/jaeger/tree/343fcc6ddd3bdb94dec864bddc2ff8794d6666a8/cmd/collector/app/sanitizer
[2]: https://github.com/jaegertracing/jaeger/blob/343fcc6ddd3bdb94dec864bddc2ff8794d6666a8/cmd/collector/app/sanitizer/zipkin/span_sanitizer.go#L169

Reply:
I can take a look at option 2.